ts/Extensions/OverlappingDataLabels.ts

```diff
@@ -268,7 +268,10 @@ Chart.prototype.hideOverlappingLabels = function (
                 box2 &&
                 label1 !== label2 && // #6465, polar chart with connectEnds
                 label1.newOpacity !== 0 &&
-                label2.newOpacity !== 0
+                label2.newOpacity !== 0 &&
+                // #15863 dataLabels are no longer hidden by translation
+                label1.attr('visibility') !== 'hidden' &&
+                label2.attr('visibility') !== 'hidden'
```

Comment:
With given O(N2) detection complexity, I would calculate and store `visibility` of the label before. Otherwise, on every iteration, we reach DOM twice (`attr('optionName')` getter does it).

Perhaps we can filter out unnecessary labels, or don't collect them at all? 
(2mins later): Just checked other labelCollectors, I think it's better to filter out them in this method, for example before sorting?